src/main/java/net/sf/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -304,26 +315,58 @@ public void saveAs() throws Throwable {
             }
         }
 
-        File oldFile = panel.getBibDatabaseContext().getDatabaseFile().orElse(null);
-        panel.getBibDatabaseContext().setDatabaseFile(file);
+        BibDatabaseContext context = panel.getBibDatabaseContext();
+
+        if (context.getLocation() == DatabaseLocation.SHARED) {
```

Comment:
This  additional code should be in some logic code (some exporter?) instead of gui.